engine/src/main/java/org/terasology/entitySystem/systems/RenderSystem.java

```diff
@@ -16,16 +16,32 @@
 package org.terasology.entitySystem.systems;
 
 /**
+ * Interface for component systems that needs to be updated every time certain game elements are rendered (generally to render something).
  */
 public interface RenderSystem extends ComponentSystem {
 
+    /**
+     * Called when opaque objects should be rendered.
+     */
     void renderOpaque();
 
+    /**
+     * Called when alpha blended objects should be rendered.
+     */
     void renderAlphaBlend();
 
+    /**
+     * Called when overlay objects should be rendered.
+     */
     void renderOverlay();
 
+    /**
+     * Called when first person objects should be rendered.
+     */
     void renderFirstPerson();
 
+    /**
+     * TODO: Implement RenderSystem.renderShadows()
```

Comment:
@emanuele3d do you have to say anything to the rendering system's defense❓ Should this part of the interface be properly documented, is it obsolete and can be removed, or what are the plans for this?